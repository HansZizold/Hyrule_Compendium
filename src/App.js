import {
  BrowserRouter as Router,
  Routes,
  Route,
} from 'react-router-dom';
import Navbar from './components/Navbar';
import Items from './components/Items';
import ItemDetailed from './components/Detail';
import ErrorPage from './components/ErrorPage';
import './App.css';

function App() {
  return (
    <Router>
      <main>
        <header>
          <Navbar />
        </header>
        <Routes>
          <Route path="/" element={<Items />} />
          <Route path="/:name/:id" element={<ItemDetailed />} />
          <Route path="*" element={<ErrorPage />} />
        </Routes>
      </main>
    </Router>
  );
}
export default App;
