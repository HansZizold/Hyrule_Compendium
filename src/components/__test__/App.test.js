import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import App from '../../App';
import store from '../../redux/configureStore';
import '@testing-library/jest-dom';

function renderWithProviders(
  ui,
  {
    storetest = store,
    ...renderOptions
  } = {},
) {
  function Wrapper({ children }) { // eslint-disable-line
    return <Provider store={storetest}>{children}</Provider>;
  }
  return { storetest, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}

test('At first load "Choose a Category" should be present', async () => {
  renderWithProviders(<App />);
  expect(screen.getByText('Choose a Category')).toBeInTheDocument();
});

describe(' App Component Rendering ', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <App />
        </Provider>,
      );
    expect(tree).toMatchSnapshot();
  });
});
