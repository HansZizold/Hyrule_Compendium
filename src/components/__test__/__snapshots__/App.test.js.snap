// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` App Component Rendering  renders correctly 1`] = `
<main>
  <header>
    <div
      className="navbar"
    >
      <img
        alt="logo"
        className="logo"
        src="dark_logo.png"
      />
      <div>
        <h1
          className="title1"
        >
          HYRULE COMPENDIUM
        </h1>
        <h2
          className="title2"
        >
          The Legend of Zelda: Breath of the Wild
        </h2>
      </div>
    </div>
  </header>
  <div>
    <div
      className="category"
    >
      <h2
        className="select"
      >
        SELECT CATEGORY
      </h2>
      <select
        className="options"
        id="category"
        name="category"
        onChange={[Function]}
      >
        <option
          className="singleoption"
          value=""
        >
          Choose a Category
        </option>
        <option
          className="singleoption"
          value="equipment"
        >
          Equipment
        </option>
        <option
          className="singleoption"
          value="creatures"
        >
          Creatures
        </option>
        <option
          className="singleoption"
          value="materials"
        >
          Materials
        </option>
        <option
          className="singleoption"
          value="monsters"
        >
          Monsters
        </option>
        <option
          className="singleoption"
          value="treasure"
        >
          Treasure
        </option>
      </select>
    </div>
    <div
      className="mainitems"
    />
  </div>
</main>
`;
